docs/kex.md

```diff
@@ -0,0 +1,131 @@
+## SSH Key Exchange
```

Comment:
This is a very good idea to document this - I always thought about it, but never got around to it. Great initiative

Reply:
It was your idea. :-)